Zend/zend_types.h

```diff
@@ -263,7 +267,7 @@ typedef struct {
 	{ NULL, (_type_mask) }
 
 #define ZEND_TYPE_INIT_CODE(code, allow_null, extra_flags) \
-	ZEND_TYPE_INIT_MASK(((code) == _IS_BOOL ? MAY_BE_BOOL : ((code) == IS_MIXED ? MAY_BE_ANY : (1 << (code)))) \
+	ZEND_TYPE_INIT_MASK(((code) == _IS_BOOL ? MAY_BE_BOOL : ( (code) == IS_ITERABLE ? _ZEND_TYPE_ITERABLE_BIT : ((code) == IS_MIXED ? MAY_BE_ANY : (1 << (code))))) \
```

Comment:
How is this supposed to work? I'd expect that you need handling in zend_register_functions to actually translate to array|Traversable.

Reply:
I really should write a test for this, not sure how to go about however.
But initially I was handling this in ``zend_check_type_slow()`` by having the following check:
```c
	if ((type_mask & _ZEND_TYPE_ITERABLE_BIT) && zend_is_iterable(arg)) {
		return 1;
	}
```
But handling this in ``zend_register_functions()`` seems smarter.

Reply:
The check is there now, so I'm assuming this is resolved?